src/viam/sdk/services/motion/client.cpp

```diff
@@ -135,6 +135,71 @@ pose_in_frame MotionClient::get_pose(
     return pose_in_frame::from_proto(response.pose());
 }
 
+void MotionClient::stop_plan(const Name& name, const AttributeMap& extra) {
+    service::motion::v1::StopPlanRequest request;
+    service::motion::v1::StopPlanResponse response;
+    grpc::ClientContext ctx;
+    set_client_ctx_authority(ctx);
+
+    *request.mutable_name() = this->name();
```

Comment:
This weirds me out. It's clearly the existing pattern on how to set things on requests -- so no objections. Out of curiosity, was this a choice we made of is this just how proto creates C++ objects?

Reply:
Yeah it's kinda weird, some fields have both a `set_foo()` method and a `mutable_foo()` method, but some only had the `mutable_foo()` method. So early on we used the `mutable_` variant more often and it just kinda stuck. It's not a super principled stance and I would not oppose taking time at some point to go through and clean these up a bit.